src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs

```diff
@@ -185,6 +186,7 @@ protected PatchResult ExecuteOnDocument(DocumentsOperationContext context, strin
                         else if (DocumentCompare.IsEqualTo(originalDoc, modifiedDocument, tryMergeMetadataConflicts: true) != DocumentCompareResult.Equal)
```

Comment:
I think it would be better to check why `DocumentCompare.IsEqual` does not return `DocumentCompareResult.NotEqual | DocumentCompareResult.AttachmentsNotEqual` flag

Reply:
I would think so too but it doesn't do what you expect it to do, if you read the implementation you could see it isn't generalized and tailored for put document and will not return attachment different for every case they are different.

Reply:
I think we need to look into that, to avoid iterating through document twice

Reply:
I already looked into it and its no good

Reply:
i'm not iterating the document i'm accessing a single property

Reply:
There seems to be wrong logic behind the compare code i'm afraid this has high probabbility to break revisions and counter code but i guess we'll need to see the test result to know